src/lib.nix

```diff
@@ -93,6 +94,7 @@
       inherit lib dlib externalSources;
       dream2nixConfig = config;
       dream2nixConfigFile = l.toFile "dream2nix-config.json" (l.toJSON config);
+      all-cabal-json = throw "all-cabal-json is not available before nixpkgs is imported";
```

Comment:
I don't think we need nixpkgs for this, nor the system. Just not passing the system should work looking at `flake-compat.nix`, since all we are really doing is using it as a source, we don't access any default outputs (which seem to be the only place where system is used).